src/plugins/WebOfTrust/util/SynchronousDelayedBackgroundJob.java

```diff
@@ -0,0 +1,231 @@
+        // deadlock.
+        if (runningJobThread == Thread.currentThread()) {
+            return;
+        }
+        // Wait for the relevant execution to complete.
+        long waitForRunCount;
+        if (runningJobThread == null) {
+            // Wait for next run to finish
+            waitForRunCount = runCount + 1;
+        } else {
+            // Wait for next run after current run to finish
+            waitForRunCount = runCount + 2;
+        }
+        while (runCount < waitForRunCount) {
+            try {
+                wait();
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                return;
+            }
+        }
+    }
+
+    /**
+     * Implementation of {@link #triggerExecution(long)} that does everything expected from that
+     * method, except it does not check for termination (the caller is assumed to have done this)
+     * and does not wait for the execution to complete.
+     * @see #triggerExecution(long)
+     * Caller must ensure synchronization on {@code this}.
+     */
+    private void triggerExecutionAsynchronously(long delay) {
+        if (delay < 0) {
+            delay = 0;
+        }
+        long newDeadline = System.currentTimeMillis() + delay;
+        if (newDeadline < nextRunDeadline) {
+            nextRunDeadline = newDeadline;
+            if (!nextRunScheduled) {
+                // Create a new waiting job thread using the given delay.
+                nextRunScheduled = true;
+                startNewJobThread();
+            } else {
+                // Notify waiting job thread that it should start earlier.
+                notifyAll();
+            }
+        }
+    }
+
+    /**
+     * Starts a new job daemon thread that:
+     * 1. Waits for the deadline to expire
+     * 2. Waits until the previous job is done, if any
+     * 3. Resets {@link #nextRunDeadline} / {@link #nextRunScheduled}
+     * 4. Runs the background job
+     * 5. Increments {@link #runCount}
+     * 6. Notifies all threads waiting upon its completion
+     * Caller must ensure synchronization on {@code this}.
+     */
+    private void startNewJobThread() {
+        Thread jobThread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                synchronized(SynchronousDelayedBackgroundJob.this) {
+                    long delay;
+                    // 1. Wait for the deadline to expire
+                    while ((delay = nextRunDeadline - System.currentTimeMillis()) > 0) {
+                        try {
+                            SynchronousDelayedBackgroundJob.this.wait(delay);
+                        } catch (InterruptedException e) {
+                            // Ignore. We can't really return here, since there is at
+                            // least one thread waiting for this thread to complete
+                            // (possibly including side effects caused by its run).
```

Comment:
I've reconsidered this, and now I do fully understand it: It is completely impossible for the class itself to have called interrupt() here, and thus it is an error and OK to ignore it.
As a consequence, I have added an explanation and a Logger.error() at commit 88c125b3abb7e962683b19e3e55fe8a851f09ae1

-> Done.